websocket.go

```diff
@@ -768,6 +834,36 @@ func (c *wsConn) handleWsConn(ctx context.Context) {
 	}
 }
 
+var onReadDeadlineResetInterval = 5 * time.Second
+
+// autoResetReader wraps a reader and resets the read deadline on if needed when doing large reads.
+func (c *wsConn) autoResetReader(reader io.Reader) io.Reader {
+	return &deadlineResetReader{
+		r:     reader,
+		reset: c.resetReadDeadline,
+
+		lastReset: time.Now(),
+	}
+}
+
+type deadlineResetReader struct {
+	r     io.Reader
+	reset func()
+
+	lastReset time.Time
+}
+
+func (r *deadlineResetReader) Read(p []byte) (n int, err error) {
+	n, err = r.r.Read(p)
+	if time.Since(r.lastReset) > onReadDeadlineResetInterval {
+		log.Warnw("slow/large read, resetting deadline while reading the frame", "since", time.Since(r.lastReset), "n", n, "err", err, "p", len(p))
```

Comment:
Should this be Warn? It isn't actionable by the user and will happen during any request which requires more than 5s to send.

Reply:
Arguably whenever any RPC takes that long to transfer, something is broken - so the user should either open an issue, or investigate their networking.